server/images.go

```diff
@@ -67,24 +67,30 @@ func (m *Model) Prompt(request api.GenerateRequest, embedding string) (string, e
 	}
 
 	var vars struct {
-		First  bool
 		System string
 		Prompt string
 		Embed  string
 
-		// deprecated: versions <= 0.0.7 used this to omit the system prompt
+		// deprecated
+		First   bool
 		Context []int
 	}
 
-	vars.First = len(request.Context) == 0
-	vars.System = m.System
+	switch {
+	case request.System != "":
+		vars.System = request.System
+
+	// only include the model's default system prompt on the
+	// first generation to avoid including it multiple times
+	case len(request.Context) == 0:
+		vars.System = m.System
+	}
+
 	vars.Prompt = request.Prompt
-	vars.Context = request.Context
 	vars.Embed = embedding
 
-	if request.System != "" {
-		vars.System = request.System
-	}
+	// deprecated
+	vars.First = true
```

Comment:
This looks like it will result in every request with old prompt templates rendering the system/instruction. Im not sure the effect this will have on response quality. That is ok if this is intentionally a breaking change.

Reply:
I think most have `{{ if and .System .First }}` but will confirm thanks for the review 😊

Reply:
Why not keep it as it since it's effectively the same thing?